mono/metadata/reflection-internals.h

```diff
@@ -41,7 +41,7 @@ MonoType *
 mono_reflection_type_handle_mono_type (MonoReflectionTypeHandle ref_type, MonoError *error);
 
 MonoType*
-mono_reflection_get_type_checked (MonoImage *rootimage, MonoImage* image, MonoTypeNameParse *info, mono_bool ignorecase, mono_bool *type_resolve, MonoError *error);
+mono_reflection_get_type_checked (MonoImage *rootimage, MonoImage* image, MonoTypeNameParse *info, mono_bool ignorecase, mono_bool *type_resolve, gboolean searchMscorlib, MonoError *error);
```

Comment:
what is the difference between mono_bool vs gboolean?

Reply:
`mono_bool` is public.
`gboolean` is private.
The choice of `int32_t` is poor though.
The type of Boolean expressions in C is `int`.
The type of Booleans in managed code is MonoBoolean -- unsigned char.
So this is a potentially third type (it could be `long`, not `int` -- it should just be `int`).

C:\s\mono2\mono\utils\mono-publib.h(88):typedef int32_t		mono_bool;
C:\s\mono2\mono\eglib\glib.h(198):typedef int32_t        gboolean;

Reply:
Using both in the same signature seems pointless, agreed.

Reply:
The only nice thing about `int32_t` here, is consistency on systems with 16bit `int`.
Anyway, no real difference, cosmetic. Usually the private names are shorter so preferred.
What to use on public signature implementations, not clear.
Maybe should rename it to `gbool` for brevity.

Reply:
That function using `mono_bool` at all is a mistake - what Jay said about public vs private usage is correct, and this is an internal function. The signature is even correct in reflection.c, so I assume the header was just an oversight in refactoring at some point. Updated.